File: Quick_Sort/quick_sort.py
def quick_sort(arr):
    if(len(arr) <= 1):
        return arr
    pivot = arr.pop()
    left = []
    right = []
    for items in arr:
        if(items > pivot):
            right.append(items)
        else:
            left.append(items)

    return quick_sort(left) + [pivot] + quick_sort(right)

my_list = [ 8, 20, -2, 4, -6 ]
print(quick_sort(my_list))

File: Algorithm/Quick_Sort/quick_sort.js
const quick_sort = (arr) => {
    if (arr.length < 2) {
        return arr;
    }
    let pivot = arr[ arr.length - 1 ];
    let left = [];
    let right = [];
    for (i = 0; i < arr.length - 1; i++) {
        if (arr[ i ] < pivot) {
            left.push(arr[ i ]);
        } else {
            right.push(arr[ i ]);
        }
    }
    return [ ...quick_sort(left), pivot, ...quick_sort(right) ];
};

const my_list = [ 8, 20, -2, 4, -6 ];
console.log(quick_sort(my_list));

// Big-O = O(nlogn)